lib/SILOptimizer/Utils/CastOptimizer.cpp

```diff
@@ -1633,6 +1633,7 @@ SILInstruction *CastOptimizer::optimizeUnconditionalCheckedCastAddrInst(
                                       StoreOwnershipQualifier::Init);
     }
     Builder.emitDestroyAddr(Loc, Inst->getSrc());
+    Builder.emitDestroyOperation(Loc, dynamicCast.getDest());
```

Comment:
Some thoughts:

1. I think it would be incorrect to emit a destroy operation here. After all, what would it mean to destroy an undef value. I am worried that we could introduce weird miscomputes.
2. Instead why don't we reuse end_lifetime? Maybe add an end_lifetime_addr?

Reply:
Done. Fixed with end_lifetime.